AWS/SecretsManager/SecretRepository.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace ToucanTech\AWS\SecretsManager;
+
+use Aws\SecretsManager\Exception\SecretsManagerException;
+use Aws\SecretsManager\SecretsManagerClient;
+
+class SecretRepository {
+
+    /**
+     * Always pin the version of AWS APIs rather than using "latest" to avoid nasty shocks

```

Comment:
:thumbsup: great advice, I would only add that fixing it  could also have a disadvantage of not getting updated or neglected over time. We could add a couple of helper comments such as:

`*
Last reviewed 23/08/2024 by XXX
TODO: Regular review and update this version periodicly to manage GoCardless stability over time
*`

If the engineers are using a TODO: tracker this may just keep it  in mind. Not critical just a suggestion.